packages/marshal/docs/copyRecord-guarantees.md

```diff
@@ -0,0 +1,57 @@
+# Why validate that an object is a CopyRecord?
+
+The input validation check `assert(passStyleOf(amount) === 'copyRecord')`, done early enough, protects against many dangers. After this test passes, we are guaranteed that `amount` is
```

Comment:
I agree with @katelynsills that this idiom needs to be shortened.  Ideally, to a name that doesn't include the technically accurate but confounding `passStyleOf` and `'copyRecord'`.

Bikeshedding a name is important for something that will be very common in our code (just as 'Far' won out over 'Remotable'), but I suggest that the body of the idiom can be:

```js
// TODO: agree on a name for this thing.
const doit = x => assert.equal(passStyleOf(harden(x)), 'copyRecord')
```

I like "pacify" and "neutralize" as names for this.

Reply:
Just to be clear, I'm suggesting two different things: a function that does the hardening + rejection of accessor properties, and helpers that check the passStyleOf. I think we should have pairs of functions, where one function modifies and the other checks, following how `Object.freeze` works: `Object.freeze` and `Object.isFrozen`, `harden` and `isHardened` (this doesn't exist as far as I know), `hardenAsCRecord/hardenAsCArray...` (doesn't yet exist) and `isCRecord/isCArray...` (doesn't yet exist)

Reply:
The reason why I don't include "passStyle" in the name is that for something like ERTP input validation which has no conception of vats or passing things remotely, it's strange to talk about when we just want to make sure that the properties of copy records are stable and we can't be attacked.

Reply:
Before resolving, added links from https://github.com/Agoric/agoric-sdk/issues/3867 and https://github.com/Agoric/agoric-sdk/issues/3885 to this conversation.